openmc/stats/multivariate.py

```diff
@@ -613,6 +616,117 @@ def from_xml_element(cls, elem):
         origin = [float(x) for x in elem.get('origin').split()]
         return cls(r, phi, z, origin=origin)
 
+class MeshSpatial(Spatial):
+    """Spatial distribution for a mesh.
+
+    This distribution specifies a mesh to sample over, chooses whether it will be volume normalized, and can set the source strengths. 
+
+    .. versionadded:: 0.13
+
+    Parameters
+    ----------
+    mesh : openmc.MeshBase
+        The mesh instance used for sampling, mesh is written into settings.xml, mesh.id is written into the source distribution
+    strengths : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+
+    Attributes
+    ----------
+    mesh : openmc.MeshBase
+        The mesh instance used for sampling, mesh is written into settings.xml, mesh.id is written into the source distribution
+    strengths : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+    """
+
+    def __init__(self, mesh, strengths=None, volume_normalized=True):
```

Comment:
If we dropped `volume_normalized` you could set the strengths to be the volumes by default if `strengths is None` which makes sense to uniformly sample by default.

Reply:
Patrick and I had a long discussion on this either on this PR or one on my fork. I think this way allows the most functionality since if you ignore volume_normalized you get the expected behavior for both passed in strengths and uniform sampling, but it allows for 4 modes of operation instead of two. Someone technically could want un-normalized passed in variables, or normalized, or uniform sampling.

Reply:
Ok I buy it, thanks @NybergWISC. I'm still a little hung up on the word `normalized` in the `volume_normalized` kwarg but as long as the docstring clearly explains its use (that it multiplies the strengths by element volumes) I'm fine with it.

Reply:
I also agree there. I originally used "volume_weighting", but we wanted to avoid using weights as it could be confused for VR rather than sampling

Reply:
One alternate possibility is `scale_by_volumes=True` but this is relatively small potatoes.